Add item query to use code gen

joshxg opened the pull request:
This is the last of the query shifting.

This has ended up being so much bigger than I thought, sorry.

From here I would like to:
1. Create a more common way of creating the sdk/api object. I am thinking to make a `context` which the host can create the `sdk/api` object on load, and then each package can use a `useApi` hook. This also has the added bonus of allow the host to inject the credentials into the `api` object so each package doesn't need to worry about it (also the URL)
2. Sort the sorting out. I think if I get to it soon, I'm just going to add a `sortKey` to the `column` - I don't want to wait..
3. Organise the different enums and use them a bit better in our code - i.e. the statuses
4. Sort the mapping out. I'm sure there's a better pattern to be using here to map things, taking away the `nodes`..
5. Organise the errors- uh oh - betterer. '
6. Add as much as I can to the graphql schema of things we want
7. I know there's a better way to be generating with multiple packages in the yaml config, but it's going to take me some time to figure it out. I think it might be best to just skip this step.
8. Sort out the mutations. I can probably dig into remote server and find the schema and add it to ours
9. Finally, I think I might PR the graphql schema into the remote server repo. Unsure on this step though..

mark-prins commented on packages/common/src/operations.graphql:
```diff
@@ -178,3 +178,88 @@ query names(
+              sellPricePerPack
+              storeId
+              totalNumberOfPacks
+              onHold
```
I had to add `name` here, and to the `StockLineNode` in the schema: the Item query below as insisting that there should be a `name` on each available batch node

mark-prins commented on packages/system/src/Item/ListView/ListView.tsx:
```diff
@@ -27,64 +31,33 @@ const listQueryFn = async ({
+      ...item,
+      availableQuantity: 0,
+      unit: '',
+      availableBatches:
```
Nice! 

There's another use of the item list unfortunately - the add item modal has another version of the query and assigning an item throws an error. My solution is to export this query and use it in that modal.. but also requires updating of the `useListData` hook to remove the page size, otherwise only 20 items show!

mark-prins reviewed (approved):
The add item modal is suffering with the change - I can update that if you'd rather just merge this and move on 😃

joshxg commented:
Ugh! Sorry! Too much of a rush before going to pick up ada 🤣 

I fixed up the query by just copy and pasting.. 🤔 Will look more closely when moving all the item things to system/item..

As for the `name`.. it doesnt seem to be used anywhere, so I just removed it 🤷
